Add troubleshooting doc for missing tier (#87526)
elastic/elasticsearch: docs/reference/troubleshooting/data/add-tier.asciidoc

[[add-tier]]
== Add a missing tier to the system

> The allocation of data in an {es} deployment can be targeted towards <<data-tiers, data tiers>>.

Reviewer: ```suggestion
The allocation of indices in an {es} deployment can be allocated on <<data-tiers, data tiers>>.
```

> In order to allow data to be allocated follow these steps to add the <<data-tiers, data tier>>
> they target to your deployment:

Reviewer: ```suggestion
In order to allow indices to be allocated, follow these steps to add the <<data-tiers, data tier>>
the indices expect to be allocated on to your deployment:
```
